pandas/core/frame.py

```diff
@@ -6827,11 +6827,23 @@ def _dispatch_frame_op(self, right, func: Callable, axis: int | None = None):
         -------
         DataFrame
         """
+        import pandas.core.computation.expressions as expressions
+
         # Get the appropriate array-op to apply to each column/block's values.
         array_op = ops.get_array_op(func)
 
         right = lib.item_from_zerodim(right)
         if not is_list_like(right):
+
+            if isinstance(self._mgr, ArrayManager):
+                use_numexpr = expressions.can_use_numexpr(
+                    func, self.shape[0], None, right
+                )
+            else:
+                use_numexpr = expressions.can_use_numexpr(func, None, None, right)
```

Comment:
why no self.shape[0]?

Reply:
Because for BlockManager, we don't know the size of the blocks at this point (numexpr is used on the block level, so it's the size of the full block and not only the size of a single column that matters)